### ISSUE TITLE
`accessibility.signalOptions.volume` isn't respected

### ISSUE THREAD

**Issue by claycoleman** [microsoft/vscode#221638]
Type: <b>Bug</b>

1. Toggle screen ready accessiblity mode to be on. Look for screen reader optimized at the bottom bar of VS Code.
2. Open settings, search for "accessibility volume". Set it to 0.
3. Take an action in a VS Code editor that creates a sound (which you haven't configured to be "off"/"never"), eg trigger an inline copilot suggestion.
4. Hear a sound (make sure your computer audio is on) even though you set volume to 0.

VS Code version: Code 1.91.1 (f1e16e1e6214d7c44d078b1f0607b2388f29d729, 2024-07-09T22:07:46.768Z)
OS version: Darwin arm64 23.5.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Pro (10 x 2400)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|6, 5, 4|
|Memory (System)|32.00GB (0.46GB free)|
|Process Argv|--crash-reporter-id 1b7c0ac2-dee9-4f37-bbaf-c0493a112d7c|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (71)</summary>

Extension|Author (truncated)|Version
---|---|---
better-comments|aar|3.0.2
quitcontrol-vscode|art|4.0.0
solarized-light-custom|bbr|4.0.1
solidjs|ber|3.0.7
vscode-tailwindcss|bra|0.12.5
turbo-console-log|Cha|2.10.4
ruff|cha|2024.30.0
regex|chr|0.4.0
doxdocgen|csc|1.4.0
vscode-eslint|dba|3.0.10
xml|Dot|2.5.1
es7-react-js-snippets|dsz|3.1.1
gitlens|eam|15.2.0
prettier-vscode|esb|10.4.0
figma-vscode-extension|fig|0.3.5
auto-rename-tag|for|0.1.10
shell-format|fox|7.2.5
kotlin|fwc|0.2.34
macros|ged|1.2.1
copilot|Git|1.213.0
copilot-chat|Git|0.17.2024062801
vscode-pull-request-github|Git|0.92.0
go|gol|0.41.4
gti-vscode|Gra|0.6.1
vscode-graphql-syntax|Gra|1.3.6
better-cpp-syntax|jef|1.27.1
vscode-go-template|jin|0.2.1
cmake-language-support-vscode|jos|0.0.9
Kotlin|mat|1.7.1
git-graph|mhu|1.30.0
vscode-docker|ms-|1.29.1
vscode-dotnet-runtime|ms-|2.0.9
black-formatter|ms-|2024.3.11501016
debugpy|ms-|2024.9.11931013
isort|ms-|2023.10.1
python|ms-|2024.10.0
vscode-pylance|ms-|2024.7.1
jupyter|ms-|2024.6.0
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.18
vscode-jupyter-cell-tags|ms-|0.1.9
vscode-jupyter-slideshow|ms-|0.1.6
remote-containers|ms-|0.375.1
remote-ssh|ms-|0.112.0
remote-ssh-edit|ms-|0.86.0
remote-wsl|ms-|0.88.2
vscode-remote-extensionpack|ms-|0.25.0
cmake-tools|ms-|1.18.43
cpptools|ms-|1.20.5
cpptools-extension-pack|ms-|1.3.0
remote-explorer|ms-|0.4.3
remote-server|ms-|1.5.2
test-adapter-converter|ms-|0.1.9
vsliveshare|ms-|1.0.5932
vsix-viewer|onl|1.0.5
vscode-hack|pra|2.20.0
quicktype|qui|23.0.170
webgl-glsl-editor|rac|1.3.8
code-spell-checker|str|3.0.1
tauri-vscode|tau|0.2.6
blueprint|tea|3.0.0
best-dark-themes-pack|the|1.0.2
best-light-themes-pack|the|1.0.2
cmake|twx|0.0.17
vscode-lldb|vad|1.10.0
vscode-swiftformat|vkn|1.6.10
change-case|wma|1.0.0
luahelper|yin|0.2.28
pretty-ts-errors|Yoa|0.5.4
markdown-all-in-one|yzh|3.6.2
material-theme|zhu|3.17.2

(28 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscoreces:30445986
vscod805:30301674
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
vscaat:30438848
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonnoceb:30805159
asynctok:30898717
pythonregdiag2:30936856
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
accentitlementst:30995554
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
chatpanelt:31048053
dsvsc021:30996838
0ee40948:31013168
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
dwnewjupyter:31046869
newcmakeconfigv2:31071590
impr_priority:31094925
refactorc:31093864
pythonrstrctxtcf:31093870

```

</details>

**Comment by claycoleman** [comment]
thanks for taking a look @meganrogge ! i believe this will resolve https://github.com/microsoft/vscode/issues/207199 as well

**Comment by Arnavion** [comment]
Specifically the setting doesn't get saved to settings.json. Even if you write it manually in settings.json, VSCode removes it on next start. Both `"accessibility.signalOptions.volume": 0` and `"accessibility.signalOptions": { "volume": 0 }` have this problem.
